Not matching on colorized clippy lints, is this expected?

Thanks for this awesome GitHub Action!

I started using it in my project somewhat recently, and I was wondering if the action is expected to match on clippy lints as well as rustc lints and test errors.

For example: I added some dead code on purpose in [this PR](https://github.com/obi1kenobi/trustfall/pull/354) and the lint job (which [uses this action](https://github.com/obi1kenobi/trustfall/blob/main/.github/workflows/ci.yml#L156-L167)) [failed as expected](https://github.com/obi1kenobi/trustfall/actions/runs/5490852131/jobs/10006790730?pr=354). However, I didn't see any problem matcher output, so either I have misconfigured something, or something else is going wrong.

This isn't terribly high-priority for me, so no rush. I appreciate any help you're able to offer!

Huh, I'm seeing a similar issue but _with_ `CARGO_TERM_COLOR=never`: Using stable clippy, this matcher does not find the line number message ([see](https://github.com/boinkor-net/ci-baseline-rust/actions/runs/5777139499/job/15656979983#step:5:11)). Looks like clippy emits a `##[error]` prefix to my messages, which seems to not match the second line in the rust-compiler matcher. Checking the raw logs, it looks like the extra context line is this:

```
2023-08-06T14:23:50.3949805Z ##[error] --> src/main.rs:3:12
```

...but maybe that's something that github actions inserts? I've played with the regex in my fork at https://github.com/antifuchs/rust-problem-matchers/tree/clippy-regex, but couldn't find a regex that gives the problem matcher the ability to extract the file/line number/column from the clippy message.

--- a/.github/matchers.json
+++ b/.github/matchers.json
@@ -4,13 +4,13 @@
             "owner": "rust-compiler",
             "pattern": [
                 {
-                    "regexp": "^(warning|warn|error)(\\[(\\S*)\\])?: (.*)$",
+                    "regexp": "^(?:\\x1B\\[[0-9;]*[a-zA-Z])*(warning|warn|error)(\\[(\\S*)\\])?(?:\\x1B\\[[0-9;]*[a-zA-Z])*: (.*?)(?:\\x1B\\[[0-9;]*[a-zA-Z])*$",
                     "severity": 1,
                     "message": 4,
                     "code": 3
                 },
                 {
-                    "regexp": "^\\s+-->\\s(\\S+):(\\d+):(\\d+)$",
+                    "regexp": "^(?:\\x1B\\[[0-9;]*[a-zA-Z])*\\s+(?:\\x1B\\[[0-9;]*[a-zA-Z])*-->\\s(?:\\x1B\\[[0-9;]*[a-zA-Z])*(\\S+):(\\d+):(\\d+)(?:\\x1B\\[[0-9;]*[a-zA-Z])*$",
                     "file": 1,
                     "line": 2,
                     "column": 3

